scripts/events_module/short/handle_short_events.py

```diff
@@ -776,89 +777,69 @@ def handle_herb_supply(self, block):
             if supply_type == "any_herb":
                 possible_herbs = []
-                for herb in herbs:
+                for herb in herb_supply.entire_supply:
                     if "always" in trigger:
                         possible_herbs.append(herb)
-                    if "low" in trigger and herbs[herb] < needed_amount / 2:
-                        possible_herbs.append(herb)
-                    if (
-                        "adequate" in trigger
-                        and needed_amount / 2 < herbs[herb] < needed_amount
-                    ):
-                        possible_herbs.append(herb)
-                    if (
-                        "full" in trigger
-                        and needed_amount < herbs[herb] < needed_amount * 2
-                    ):
-                        possible_herbs.append(herb)
-                    if "excess" in trigger and needed_amount * 2 < herbs[herb]:
+
+                    rating = herb_supply.get_herb_rating(herb)
+                    if rating in trigger:
                         possible_herbs.append(herb)
+
                 self.chosen_herb = random.choice(possible_herbs)
 
             # if it wasn't a random herb or all herbs, then it's one specific herb
             else:
                 self.chosen_herb = supply_type
 
+            herb_list.append(self.chosen_herb)
+
             # now adjust the supply for the chosen_herb
+            total_herb = herb_supply.total_of_herb(self.chosen_herb)
             if adjustment == "reduce_full":
-                herbs[self.chosen_herb] = 0
+                herb_supply.remove_herb(self.chosen_herb, total_herb)
             elif adjustment == "reduce_half":
-                herbs[self.chosen_herb] = int(game.clan.herbs[self.chosen_herb] / 2)
+                herb_supply.remove_herb(self.chosen_herb, total_herb / 2)
             elif adjustment == "reduce_quarter":
-                herbs[self.chosen_herb] = int(game.clan.herbs[self.chosen_herb] / 4)
+                herb_supply.remove_herb(self.chosen_herb, total_herb / 4)
             elif adjustment == "reduce_eighth":
-                herbs[self.chosen_herb] = int(game.clan.herbs[self.chosen_herb] / 8)
+                herb_supply.remove_herb(self.chosen_herb, total_herb / 8)
             elif "increase" in adjustment:
-                herbs[self.chosen_herb] += int(adjustment.split("_")[1])
-
-        if not self.chosen_herb:
-            self.chosen_herb = random.choice(list(herbs.keys()))
-        if self.chosen_herb:
-            herb_list.append(self.chosen_herb)
-
-        if herb_list:
-            for herb in herb_list:
-                if herb in herbs and herbs[herb] == 0:
-                    herbs.pop(herb)
+                herb_supply.add_herb(self.chosen_herb, int(adjustment.split("_")[1]))
 
-        self.herb_notice = self.herb_notice + adjust_list_text(herb_list)
+        self.herb_notice = self.herb_notice + adjust_list_text(herb_list) + "during this moon."
```

Comment:
Localization! \*bites u\*
[This might help you figure out how it works](https://pygame-gui.readthedocs.io/en/latest/localization.html)